### ISSUE TITLE
mellon/login?ReturnTo 303 Redirect is it necessary?

### ISSUE THREAD

**Issue by gsustek** [latchset/mod_auth_mellon#17]
Hi, would you be so kind to explain me is it possible not to have HTTP 303 redirect after HTTP GET to some resource protected by auth_mellon, but instead to just send HTT200 with autosubmit form in Response. Shibboleth and other SAML SSO SP does  in this way. Why is this different, can it be configured not to do 303 redirect?
![image](https://user-images.githubusercontent.com/5320033/78084329-25963880-73b8-11ea-8ba9-e2b6d70b1ecd.png)
Shibboleth;
![image](https://user-images.githubusercontent.com/5320033/78084419-6ee68800-73b8-11ea-9bac-e56fa0438f42.png)
Shibboleth Response header
![image](https://user-images.githubusercontent.com/5320033/78084435-7c037700-73b8-11ea-8d61-582c7be5b1af.png)

Kind Regards,
Goran.

**Comment by thijskh** [comment]
It does do this for me just fine. Perhaps you've configured the IdP metadata to use HTTP-POST binding for the authnrequest?

**Comment by gsustek** [comment]
yes, exactly, this is my intention, to do HTTP-POST binding for authnrequest. And you can see from pictures above that in that case, mellon has this redirect to it self mellon/login?Return, and shibboleth doesn't have it. Is there some standard how to do that, or just a matter of implementation?
Can i "hide" it with mode_rewrite RewriteRule somehow? Thnx.

**Comment by thijskh** [comment]
Ah, sorry for misunderstanding. Yes, this should be supported. Although it's uncommon to do so in practice - HTTP-Redirect binding is the standard way to do it.

Do you have the HTTP-POST binding correctly configured in the IdP metadata that you supplied to Mellon?

**Comment by gsustek** [comment]
Yes, i have SAML POST binding with authnrequest and authnresponse. configured. Here are
[Xml-s.zip](https://github.com/latchset/mod_auth_mellon/files/4416282/Xml-s.zip)

 my xml-s for IDP and SP.

**Comment by gsustek** [comment]
Workaround with this RewriteRule:

    RewriteCond "%{HTTP_REFERER}" "!idp.example.com"
    RewriteRule "^(.*)$" "/mellon/login?ReturnTo=%{REQUEST_SCHEME}://%{HTTP_HOST}%{REQUEST_URI}&IdP=http://idp.example.com:8080/auth/realms/test_realm" [PT]


but a fix or a "feature" would be so welcome:-)

**Comment by CHAORYU** [comment]
I'm working with keystone federtation function and use the mod_auth_mellon,Why when post to .../mellon/postResponse it returns to 303 but not set cookie in my situation.
